pkg/handlers/generic/mutation/extraapiservercertsans/inject.go

```diff
@@ -50,12 +61,16 @@ func (h *extraAPIServerCertSANsPatchHandler) Mutate(
 	obj *unstructured.Unstructured,
 	vars map[string]apiextensionsv1.JSON,
 	holderRef runtimehooksv1.HolderReference,
-	_ client.ObjectKey,
+	clusterKey ctrlclient.ObjectKey,
 ) error {
 	log := ctrl.LoggerFrom(ctx).WithValues(
 		"holderRef", holderRef,
 	)
-
+	cluster := &capiv1.Cluster{}
+	if err := h.client.Get(ctx, clusterKey, cluster); err != nil {
+		return err
+	}
+	defaultAPICertSANs := getDefaultAPIServerSANs(cluster)
```

Comment:
WDYT leaving a TODO?
```suggestion
        TODO: move the defaults to a mutating webhook
	defaultAPICertSANs := getDefaultAPIServerSANs(cluster)
```